src/gtfs/read.rs

```diff
@@ -445,24 +494,212 @@ fn get_lines_from_gtfs(gtfs_routes: &[Route], read_mode: &RouteReadType) -> Vec<
+                for r2 in iter.clone() {
+                    if r1.is_same_line(r2) {
+                        backward_routes.insert(r2);
+                        if r1.id > r2.id {
+                            routes.remove(r1);
+                            routes.insert(r2);
+                        }
+                    } else {
+                        routes.insert(r2);
+                    }
+                }
+            }
+
+            for r in routes {
+                let l = objects::Line {
+                    id: r.id.to_string(),
+                    code: line_code(r),
+                    codes: KeysValues::default(),
+                    object_properties: KeysValues::default(),
+                    comment_links: CommentLinksT::default(),
+                    name: r.long_name.to_string(),
+                    forward_name: None,
+                    forward_direction: None,
+                    backward_name: None,
+                    backward_direction: None,
+                    color: r.color.clone(),
+                    text_color: r.text_color.clone(),
+                    sort_order: r.sort_order,
+                    network_id: line_agency(r),
+                    commercial_mode_id: r.route_type.to_gtfs_value(),
+                    geometry_id: None,
+                    opening_time: None,
+                    closing_time: None,
+                };
+                lines.push(l);
+            }
         }
     }
     lines
 }
 
-pub fn read_routes<P: AsRef<path::Path>>(path: P, collections: &mut Collections) {
-    let path = path.as_ref().join("routes.txt");
-    let mut rdr = csv::Reader::from_path(path).unwrap();
-    let gtfs_routes: Vec<Route> = rdr.deserialize().map(Result::unwrap).collect();
+fn get_routes_from_gtfs(
+    gtfs_routes: &[Route],
+    gtfs_trips: &[Trip],
+    read_mode: &RouteReadType,
+) -> Vec<objects::Route> {
+    let mut routes = vec![];
+
+    match *read_mode {
+        RouteReadType::RouteAsNtmLine => {
+            let backward_routes: HashSet<_> = gtfs_trips
+                .iter()
+                .filter(|t| t.direction == DirectionType::Backward)
+                .map(|t| t.route_id.clone())
+                .collect();
+            for r in gtfs_routes {
+                let route = objects::Route {
+                    id: r.id.to_string(),
+                    name: r.long_name.to_string(),
+                    direction_type: Some("forward".to_string()),
+                    codes: KeysValues::default(),
+                    object_properties: KeysValues::default(),
+                    comment_links: CommentLinksT::default(),
+                    line_id: r.id.to_string(),
+                    geometry_id: None,
+                    destination_id: None,
+                };
+                routes.push(route);
+                if backward_routes.contains(&r.id) {
+                    let route = objects::Route {
+                        id: r.id.to_string() + "_R",
```

Comment:
what about clash?

Reply:
clash ? the rock band? I don't understand